Tribler/Test/Community/Trustchain/test_community.py

```diff
@@ -320,3 +357,154 @@ def test_crawl_batch(self):
+                                                                  u"unknown")
+        candidate.associate(other.community.my_member)
+        candidate.walk_response(time.time())
+
+        # Assert
+        intro = blockingCallFromThread(reactor, node.community.dispersy_get_introduce_candidate,
+                                       node.my_candidate)
+        self.assertIsNotNone(intro)
+        self.assertIsInstance(intro, Candidate)
+        self.assertEqual(intro, candidate)
+
+    def test_live_edge_recommend_valid(self):
+        """
+        Live edges should never include invalid/old candidates.
+        """
+        # Arrange
+        node, other, another = self.create_nodes(3)
+
+        # Stop the community from walking/crawling once it gets reactor control
+        node.community.cancel_all_pending_tasks()
+        node.community.reset_live_edges()
+        node.community.candidates.clear()
+
+        candidate = node.community.create_or_update_walkcandidate(other.my_candidate.sock_addr,
+                                                                  other.my_candidate.sock_addr,
+                                                                  ('0.0.0.0', 0),
+                                                                  other.my_candidate.tunnel,
+                                                                  u"unknown")
+        candidate.associate(other.community.my_member)
+        candidate.walk_response(time.time())
+
+        node.community.create_or_update_walkcandidate(another.my_candidate.sock_addr,
+                                                      another.my_candidate.sock_addr,
+                                                      ('0.0.0.0', 0),
+                                                      another.my_candidate.tunnel,
+                                                      u"unknown")
+
+        # Assert
+        intro = blockingCallFromThread(reactor, node.community.dispersy_get_introduce_candidate,
+                                       node.my_candidate)
+        self.assertIsNotNone(intro)
+        self.assertIsInstance(intro, Candidate)
+        self.assertEqual(intro, candidate)
+
+    def test_live_edge_callback_no_candidates(self):
+        """
+        Test live edges start with my member.
+        """
+        # Arrange
+        node, = self.create_nodes(1)
+
+        called = [False]
+
+        def check_live_edge(edge_id, candidates):
+            self.assertEqual(1, edge_id)
+            self.assertEqual(node.my_member.mid, candidates[0].get_member().mid)
+            called[0] = True
+
+        node.community.set_live_edge_callback(check_live_edge)
+
+        # Stop the community from walking/crawling once it gets reactor control
+        node.community.cancel_all_pending_tasks()
+        node.community.reset_live_edges()
+
+        # Act
+        node.community.take_step()
+
+        # Assert
+        self.assertTrue(called)
```

Comment:
This is not valid I think, `called` will always be true, regardless whether `called[0]` is true or false.